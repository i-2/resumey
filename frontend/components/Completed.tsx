import * as React from 'react';
import { Text } from '@chakra-ui/react';

export default (props: any) => <Text> Completed !! </Text>