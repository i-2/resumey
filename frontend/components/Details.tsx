import React from 'react';
import {
    Flex, Button, FormControl, FormLabel, Input, FormErrorMessage, Wrap, WrapItem
} from '@chakra-ui/react';
import {
    Field,
    Formik
} from 'formik';
import * as Yup from 'yup';

export interface ValueProps {
    name: string,
    title: string,
    validate?: any
}

export interface DetailProps {
    schema: ValueProps[],
    initValues: any,
    onNext: () => any,
    submitValues: (value: any) => any
    onPrevious?: () => any
}

function getValues(schema: ValueProps[]) {
    let values: any = {};
    for (let value of schema) {
        values[value.name] = "";
    }
    return values;
}

function getValidation(schema: ValueProps[]) {
    let values: any = {};
    for (let value of schema) {
        values[value.name] = value.validate;
    }
    return Yup.object().shape(values);
}


export function renderField(values: ValueProps[]) {
    return (<Wrap width="100%" borderWidth="1px" borderRadius="lg" padding="5%" marginTop="2%">
        {
            values.map((value: any, i: number) =>
                <WrapItem key={i} width="45%" margin="10px">
                    <Field name={value.name}>
                        {({ field, form }: any) => (
                            <FormControl isInvalid={form.errors[value.name] && form.touched[value.name]}>
                                <FormLabel htmlFor={value[value.name]}> {value.title} </FormLabel>
                                <Input {...field}
                                    id={value[value.name]}
                                    placeholder={value.title}
                                />
                                <FormErrorMessage>{form.errors[value.name]}</FormErrorMessage>
                            </FormControl>
                        )}
                    </Field>
                </WrapItem>)}
    </Wrap>)
}



export const Details = (props: DetailProps) => {

    return (<Flex width="100%" direction="row" justify="center">
        <Flex width="100%" direction="column" justify="center">
            <Formik
                initialValues={props.initValues}
                validationSchema={getValidation(props.schema)}
                onSubmit={(values, actions) => {
                    props.submitValues(values);
                    props.onNext();
                }}>
                {
                    (innerProps: any) => (
                        <form style={{ width: "100%" }} onSubmit={innerProps.handleSubmit}>
                            <Flex width="100%" direction="row" justify="space-between">
                                {renderField(props.schema)}
                            </Flex>
                            {
                                props.onPrevious ? <Button
                                    mt={4}
                                    colorScheme="teal"
                                    type="submit"
                                    m="5"
                                    onClick={props.onPrevious}
                                >
                                    Previous </Button> : <div></div>
                            }
                            <Button
                                mt={4}
                                colorScheme="teal"
                                type="submit"
                                m="5"
                            >
                                Next
                            </Button>


                        </form>
                    )
                }
            </Formik>
        </Flex>
    </Flex >)
}