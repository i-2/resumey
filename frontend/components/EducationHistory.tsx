import React from 'react';
import { ArrayForm } from './chakraForms/ArrayForm';
import * as Yup from 'yup';


const EDUCATION_FIELDS = [
    { name: "school", title: "School", placeholder: "Enter the name of school", "type": "input", "validate": Yup.string().required() },
    { name: "degree", title: "Degree", placeholder: "Enter the name of degree", "type": "input", "validate": Yup.string().required() },
    { name: "start", title: "Start Date", placeholder: "Enter the start date", "type": "date", "validate": Yup.date().required() },
    { name: "end", title: "End Date", placeholder: "Enter the end date", "type": "date", "validate": Yup.date().required() },
    { name: "city", title: "City", placeholder: "Enter the place of study", "type": "input", "validate": Yup.string().required()},
    { name: "country", title: "Country", placeholder: "Enter the country of country", "type": "input", "validate": Yup.string().required()},
    { name: "description", title: "Description", placeholder: "Care to explain ?", "type": "text", "validate": Yup.string().required() }
]


export const EducationHistory = (props: any) => <ArrayForm
    name="education"
    fields={EDUCATION_FIELDS}
    initValues={{ "education": props.education }}
    onPrevious={props.onPrevious}
    onAddMore={props.onAddMore}
    onSubmit={props.onSave}
    onNext={props.onNext}
/>