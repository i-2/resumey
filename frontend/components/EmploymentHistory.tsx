import React from 'react';
import { ArrayForm } from './chakraForms/ArrayForm';


const EMPLOYMENT_FIELDS = [
    { name: "employer", title: "Employer", placeholder: "Enter the name of employer", "type": "input" },
    { name: "designation", title: "Designation", placeholder: "Enter the designation", "type": "input" },
    { name: "start", title: "Start Date", placeholder: "Enter the start date", "type": "date" },
    { name: "end", title: "End Date", placeholder: "Enter the end date", "type": "date" },
    { name: "summary", title: "Summary", placeholder: "Summary", "type": "text" }
]


export const EmploymentHistory = (props: any) => <ArrayForm
    name="history"
    fields={EMPLOYMENT_FIELDS}
    initValues={{"history": props.history}}
    onPrevious={props.onPrevious}
    onAddMore={props.onAddMore}
/>