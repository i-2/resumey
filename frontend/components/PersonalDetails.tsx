import React from 'react';
import {
    Flex, Button, FormControl, FormLabel, Input, FormErrorMessage, Text, Wrap, WrapItem
} from '@chakra-ui/react';
import {
    Form,
    Field,
    Formik,
} from 'formik';


export interface ValueProps {
    name: string,
    title: string
}

export interface PersonalDetailProps {
    schema: ValueProps[],
    onNext: () => any;
}


export function renderField(values: ValueProps[]) {
    return (<Wrap width="100%">
        {values.map((value: any, i: number) => <WrapItem key={i} width="45%" margin="10px"><Field name={value.name}>
            {({ field, form }: any) => (
                <FormControl isInvalid={form.errors.name && form.touched.name}>
                    <FormLabel htmlFor={value.name}> {value.title} </FormLabel>
                    <Input {...field} id={value.name} placeholder={value.title} />
                    <FormErrorMessage>{form.errors.name}</FormErrorMessage>
                </FormControl>
            )}
        </Field></WrapItem>)}
    </Wrap>)
}



export const PersonalDetails = (props: PersonalDetailProps) => (
    <Flex width="100%" direction="row" justify="center">
        <Formik
            initialValues={{}}
            onSubmit={(values, { setSubmitting }) => { }}>
            <Flex width="100%" direction="column" justify="center">
                <Form style={{ width: "100%" }}>
                    <Flex width="100%" direction="row" justify="space-between">
                        {renderField(props.schema)}
                    </Flex>
                    <Button
                        mt={4}
                        colorScheme="teal"
                        isLoading={false}
                        type="submit"
                        m="5"
                        onClick={props.onNext}
                    >
                        Next
                        </Button>

                </Form>
            </Flex>
        </Formik>
    </Flex>


)