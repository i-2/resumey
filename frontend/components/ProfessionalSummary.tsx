import React from 'react';
import {
    Flex, FormControl, FormLabel, Textarea, FormErrorMessage, Button
} from '@chakra-ui/react';
import {
    Field,
    Formik,
} from 'formik';
import * as Yup from 'yup';

const SummarySchema = Yup.object().shape(
    { "summary": Yup.string().required() }
)

export interface ProfessionalSummaryProps {
    name: string,
    title: string,
    initValues: any,
    onNext: () => any,
    onPrev: () => any,
    submitValues: (values: any) => any
}

export function ProfessionalSummary(props: ProfessionalSummaryProps) {
    const { name, title } = props;
    return (

        <Flex width="100%" direction="row" justify="center">
            <Formik
                initialValues={props.initValues}
                validationSchema={SummarySchema}
                onSubmit={(values, actions) => {
                    props.onNext();
                    props.submitValues(values);
                }}>
                {
                    (innerProps: any) => (
                        <Flex width="100%" direction="column" justify="center">
                            <form style={{ width: "100%" }} onSubmit={innerProps.handleSubmit}>
                                <Field name={name}>
                                    {({ field, form }: any) => {

                                        return (

                                            <FormControl isInvalid={form.errors.summary && form.touched.summary}>
                                                <FormLabel htmlFor={name}> {title} </FormLabel>
                                                <Textarea placeholder="Enter your professional summary"  {...field} />
                                                <FormErrorMessage>{form.errors.summary}</FormErrorMessage>
                                            </FormControl>

                                        )
                                    }
                                    }
                                </Field>
                                <Button
                                    mt={4}
                                    colorScheme="teal"
                                    isLoading={false}
                                    onClick={props.onPrev}
                                    m="5"
                                >
                                    Previous
                                </Button>
                                <Button
                                    mt={4}
                                    colorScheme="teal"
                                    isLoading={false}
                                    type="submit"
                                    m="5"
                                >
                                    Next
                                </Button>
                            </form>
                        </Flex>
                    )
                }

            </Formik>
        </Flex>

    );
}