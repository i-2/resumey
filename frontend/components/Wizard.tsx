import React from 'react';
import {
    Flex, Text
} from '@chakra-ui/react';
import { Component } from 'react';
import { ProgressBar } from './ProgressBar';
import Completed from './Completed';



export interface WizardItem {
    title: string,
    component: Component
}

export interface WizardProps {
    tabs: WizardItem[]
    count: number
}

export const Wizard = (props: WizardProps) => {
    if (props.count < props.tabs.length) {
        const X = props.tabs[props.count];
    }
    else {
        const X = { "title": "Completed", "component": <Completed /> }
    }
    const percentage = Math.floor((props.count / props.tabs.length) * 100);
    return (
        <Flex width="100%" borderRadius="50" m="5" justify="center" direction="column">
            <Flex w="100%" justify="center">
                <ProgressBar value={percentage} />
            </Flex>
            {
                <Flex justify="center" direction="column">
                    <Flex direction="row" justify="center">
                        <Text fontSize="3xl"> {X.title} </Text>
                    </Flex>
                    <Flex width="100%" direction="row" justify="center">
                        {X.component}
                    </Flex>
                </Flex>

            }
        </Flex>
    )
}




