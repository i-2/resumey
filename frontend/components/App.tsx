import React from 'react';
import { Flex, Text } from '@chakra-ui/react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as actionCreators from '../actions';
import { Details } from './Details';
import { ProfessionalSummary } from './ProfessionalSummary';
import { Wizard } from './Wizard';
import * as Yup from 'yup';

const PERSONAL_DETAIL_SCHEMA: any[] = [
    { "title": "First Name", "name": "firstName", "validate": Yup.string().required() },
    { "title": "Last Name", "name": "lastName", "validate": Yup.string().required() },
    { "title": "Email", "name": "email", "validate": Yup.string().email().required() },
    { "title": "Phone Number", "name": "phone", "validate": Yup.string().required() },
    { "title": "Country", "name": "country", "validate": Yup.string().required() },
    { "title": "City", "name": "city", "validate": Yup.string().required() },
    { "title": "Address", "name": "address", "validate": Yup.string().required() },
    { "title": "Postal Code", "name": "pinCode", "validate": Yup.string().required() }
]


export const _App = (props: any) => (
    <Flex direction="column"
        align="center"
        maxW={{ xl: "1200px" }}
        m="0 auto">
        <Flex w="100%" justify="center">
            <Text fontSize="3xl"> Resume Builder </Text>
        </Flex>
        <Wizard tabs={
            [
                {
                    "title": "Personal Details", "component": <Details
                        schema={PERSONAL_DETAIL_SCHEMA}
                        values={props.values}
                        onNext={() => props.updateWizard(1)}
                        onDetailChange={(name, value) => {
                            props.updatePersonalDetail(name, value);
                        }}
                    />
                },
                { "title": "Professional Summary", "component": <ProfessionalSummary title="Summary" name="summary" onNext={() => props.updateWizard(2)} onPrev={() => props.updateWizard(0)} /> }
            ]
        } count={props.count} />
    </Flex>
)

const mapStateToProps = (state: any) => ({ count: state.wizard.windowCount, values: state.personalDetail })
const mapDispatchToProps = (dispatch: any) => bindActionCreators(actionCreators, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(_App);