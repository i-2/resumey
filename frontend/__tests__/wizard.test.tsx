import * as  React from 'react';
import * as renderer from 'react-test-renderer';
import {Wizard} from '../components/Wizard';

it('renders wizard correctly', () => {
  const tree = renderer
    .create(<Wizard tabs={[]} count={0}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});