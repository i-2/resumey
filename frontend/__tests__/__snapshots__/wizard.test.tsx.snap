// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders wizard correctly 1`] = `
<div
  className="css-155kcuw"
>
  <div
    className="css-ldo4d5"
  >
    <div
      className="css-13yu7w7"
    >
      <div
        className="css-k008qs"
      >
        <div
          className="css-1xet117"
        >
          Completion : 
          NaN
          %
        </div>
        <div
          className="css-17xejub"
        />
        <button
          className="css-1xet117"
        >
          <svg
            className="chakra-icon css-dvq678"
            focusable={false}
            viewBox="0 0 24 24"
          >
            <g
              stroke="currentColor"
              strokeWidth="1.5"
            >
              <path
                d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                fill="none"
                strokeLinecap="round"
              />
              <path
                d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                fill="none"
                strokeLinecap="round"
              />
              <circle
                cx="12"
                cy="12"
                fill="none"
                r="11.25"
                strokeMiterlimit="10"
              />
            </g>
          </svg>
        </button>
      </div>
      <div
        className="css-kdwx3d"
      >
        <div
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={NaN}
          className="css-10klw3m"
          role="progressbar"
          style={
            Object {
              "width": "NaN%",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="css-zg1vud"
  >
    <div
      className="css-1vp8jwk"
    >
      <p
        className="chakra-text css-1tapsqs"
      >
         
        Completed
         
      </p>
    </div>
    <div
      className="css-2l42up"
    >
      <p
        className="chakra-text css-0"
      >
         Completed !! 
      </p>
    </div>
  </div>
</div>
`;
